java-checks-test-sources/src/main/java/checks/regex/UnicodeCaseCheck.java

```diff
@@ -0,0 +1,28 @@
+package checks.regex;
+
+import java.util.regex.Pattern;
+
+public class UnicodeCaseCheck {
+
+  void noncompliant(String str) {
+    Pattern.compile("söme pättern", Pattern.CASE_INSENSITIVE); // Noncompliant [sc=37;ec=61] {{Also use "Pattern.UNICODE_CASE" to correctly handle non-ASCII letters.}}
```

Comment:
be careful, in order to test the location of the issues, you need 2 opening/closing brackets! For instance:  `[[sc=37,ec=61]]`.
I didn't check the other rule but I might have missed it, and with this PR, the precise location is not tested because of this error.

Reply:
I fixed this and also grepped through the other tests to see whether I made the same mistake elsewhere (I didn't).